lib/spack/spack/schema/ci.py

```diff
@@ -47,7 +47,7 @@
         "tags": {"type": "array", "items": {"type": "string"}},
         "variables": {
             "type": "object",
-            "patternProperties": {r"[\w\d\-_\.]+": {"type": "string"}},
+            "patternProperties": {r"[\w\d\-_\.]+": {"type": ["string", "number"]}},
```

Comment:
@scottwittenburg I have been annoyed by this for ages and this finally required changing it, any reason this shouldn't work that you remember?